lib/Dialect/SV/SVOps.cpp

```diff
@@ -198,6 +199,42 @@ void IfOp::build(OpBuilder &odsBuilder, OperationState &result, Value cond,
   }
 }
 
+/// Replaces the given op with the contents of the given single-block region,
+/// using the operands of the block terminator to replace operation results.
+static void replaceOpWithRegion(PatternRewriter &rewriter, Operation *op,
+                                Region &region, ValueRange blockArgs = {}) {
+  assert(llvm::hasSingleElement(region) && "expected single-region block");
+  Block *block = &region.front();
+  Operation *terminator = block->getTerminator();
+  ValueRange results = terminator->getOperands();
+  rewriter.mergeBlockBefore(block, op, blockArgs);
+  rewriter.replaceOp(op, results);
```

Comment:
`terminator->getOperands()` will always return an empty set of values (since yield in an if never has operands).  can you just use `rewriter.eraseOp()` ?

Reply:
Yes, I changed